client/installation-helper.cc

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright (c) 2015 Novell, Inc.
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact Novell, Inc.
+ *
+ * To contact Novell about this file by physical or electronic mail, you may
+ * find current contact information at www.novell.com.
+ */
+
+
+#include "config.h"
+
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <iostream>
+#include <boost/algorithm/string.hpp>
+
+#include <snapper/Snapper.h>
+#include <snapper/AppUtil.h>
+#include <snapper/SnapperDefines.h>
+#include <snapper/Btrfs.h>
+#include <snapper/FileUtils.h>
+#include <snapper/Hooks.h>
+
+#include "utils/GetOpts.h"
+
+
+using namespace snapper;
+using namespace std;
+
+
+void
+step1(const string& device, const string& description)
+{
+    // step runs in inst-sys
+
+    cout << "step 1 device:" << device << endl;
+
+    cout << "temporarily mounting device" << endl;
+
+    SDir s_dir("/");
+
+    TmpMount tmp_mount(s_dir, device, "tmp-mnt-XXXXXX", "btrfs", 0, "");
+
+    cout << "copying config-file" << endl;
+
+    mkdir((tmp_mount.getFullname() + "/etc").c_str(), 0777);
+    mkdir((tmp_mount.getFullname() + "/etc/snapper").c_str(), 0777);
+    mkdir((tmp_mount.getFullname() + "/etc/snapper/configs").c_str(), 0777);
+
+    try
+    {
+	SysconfigFile config(CONFIGTEMPLATEDIR "/" "default");
+
+	config.setName(tmp_mount.getFullname() + CONFIGSDIR "/" "root");
+
+	config.setValue(KEY_SUBVOLUME, "/");
+	config.setValue(KEY_FSTYPE, "btrfs");
+    }
+    catch (const FileNotFoundException& e)
+    {
+	cerr << "copying config-file failed" << endl;
```

Comment:
I do not get why it is called copying when it looks like creating new file using template and change some values for me.